odoo/addons/base/tests/test_mail.py

```diff
@@ -366,6 +368,28 @@ def test_cid_with_at(self):
     #     for ext in test_mail_examples.MSOFFICE_1_OUT:
     #         self.assertNotIn(ext, new_html)
 
+    def test_multi_quote_detection(self):
+        doc = """
+            <div>
+                <div>--<br/><span>Hello</span>
+                <div> --<br/> Odoo <br/></div>
+                <br/>My Signature </div>
+            </div>
+        """
+        doc = html.fromstring(doc)
+
+        for el in doc.iter(tag=etree.Element):
+            tag_quote(el)
+
+        res1 = doc
+
+        for el in doc.iter(tag=etree.Element):
+            tag_quote(el)
+
+        res2 = doc
+
+        self.assertEqual(res1, res2)
```

Comment:
You're never cloning the xml tree so you're just comparing the same reference, which will always succeed.

This is also a good case to check against the actual intended result as we know we don't want "Hello" to be quoted, that's the main thing we're trying to avoid above the fact that it doesn't change from one iteration to the next

Reply:
I will make modifications in test case :ok_hand: